common/unsafe/src/test/java/org/apache/spark/unsafe/types/CollationSupportSuite.java

```diff
@@ -606,7 +606,8 @@ private void assertUpper(String target, String collationName, String expected)
     UTF8String target_utf8 = UTF8String.fromString(target);
     UTF8String expected_utf8 = UTF8String.fromString(expected);
     int collationId = CollationFactory.collationNameToId(collationName);
-    assertEquals(expected_utf8, CollationSupport.Upper.exec(target_utf8, collationId));
+    assertEquals(expected_utf8, CollationSupport.Upper.exec(target_utf8, collationId, true));
+    assertEquals(expected_utf8, CollationSupport.Upper.exec(target_utf8, collationId, false));
```

Comment:
Here and below: please add a comment that we are testing here with both the JVM-based and the ICU-based version of the expressions, and that for the test cases we are testing here the results are the same.